src/site/content/en/web-dev-basics-one/fetch-api-error-handling/index.md

```diff
@@ -0,0 +1,234 @@
+---
+layout: post
+title: Implement error handling when using the Fetch API
+description: Catching errors when working with the Fetch API. 
+date: 2022-05-23
+---
+
+This article demonstrates some error handling approaches when working with the [Fetch API](https://developers.google.com/web/ilt/pwa/working-with-the-fetch-api?hl=en). The Fetch API lets you make a request to a remote network resource. When you make a remote network call, your web page becomes subject to a variety of potential network errors. 
+
+The following sections describe potential errors and prescribe how to write code that provides a sensible level of functionality that is resilient to errors and unexpected network conditions. Resilient code keeps your users happy and maintains a standard level of service for your website. 
+
+## Anticipate potential network errors
+
+This section describes a scenario in which the user creates a new video named   
+"My Travels.mp4" and then attempts to upload the video to a video-sharing website.
+
+When working with Fetch, it's easy to consider the [happy path](https://en.wikipedia.org/wiki/Happy_path) where the user successfully uploads the video. However, there are other paths that are not as smooth, but for which web developers must plan. Such (unhappy) paths can happen due to user error, through unexpected environmental conditions, or because of a bug on the video-sharing website.
+
+### Examples of user errors
+
+-  The user uploads an image file (such as JPEG) instead of a video file.
+-  The user begins uploading the wrong video file. Then, part way through the upload, the user specifies the correct video file for upload.
+-  The user accidentally clicks "Cancel upload" while the video is uploading.
+
+### Examples of environmental changes
+
+-  The internet connection goes offline while the video is uploading.
+-  The browser restarts while the video is uploading.
+-  The servers for the video-sharing website restart while the video is uploading.
+
+### Examples of errors with the video-sharing website
+
+-  The video-sharing website cannot handle a filename with a space. Instead of "My Travels.mp4", it expects a name such as "My_Travels.mp4" or "MyTravels.mp4".
```

Comment:
```suggestion
-  The video-sharing website cannot handle a filename with a space. Instead of `"My Travels.mp4"`, it expects a name such as `"My_Travels.mp4"` or `"MyTravels.mp4"`.
```